### PR TITLE
feat: add separate urls for portfolio page and public profile

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/regen-registry#1366
- add separate urls for the public and private profile page
- update `IconTabs` to handle tab with link and to support initial active tab
- use nested routes and `<Outlet />` component for both  public and private profiles to preserve the layout
- rename `App` to `routes` and clean it

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

1. https://deploy-preview-1549--regen-registry.netlify.app/

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-components/src/components/tabs/IconTab.tsx [hunk 1]
```diff
@@ -7,17 +7,20 @@ import type { Theme } from 'src/theme/muiTheme';
...
   size?: TextSize;
+  href?: string;
+  linkComponent?: LinkComponentProp;
   sxInner?: SxProps<Theme>;
 }
 
 const StyledTab = styled(Tab, {
   shouldForwardProp: prop => prop !== 'size',
-})<TabProps & { size: TextSize }>(({ theme, size }) => ({
+})<TabProps & { href?: string; size: TextSize }>(({ theme, size }) => ({
```
Comment (flagrede) [inline web-components/src/components/tabs/IconTab.tsx:23]: `href` should already be typed inside `TabProps` but for some reason, TS does not find it so I had to explicitly add it.

File: web-components/src/components/tabs/IconTabs.tsx [hunk 2]
```diff
@@ -67,23 +79,26 @@ const IconTabs: React.FC<IconTabsProps> = ({
...
-          key={index}
-          value={value}
-          index={index}
-          hidden={tab.hidden || value !== index}
-          sxs={{ ...sxs?.panel }}
-        >
-          {tab.content}
-        </TabPanel>
-      ))}
+      {hasContent &&
```
Comment (flagrede) [inline web-components/src/components/tabs/IconTabs.tsx:90]: Allow IconTabs to be used with tabs only for usage with react-router and nested routes.

File: web-registry/src/routes.tsx [hunk 3]
```diff
@@ -1,23 +1,22 @@
-import React, { lazy } from 'react';
+import { lazy } from 'react';
```
Comment (flagrede) [inline web-registry/src/routes.tsx:1]: This file has been renamed to `routes.tsx` as it only contains routes configuration now.

File: web-registry/src/components/organisms/Portfolio.tsx [hunk 4]
```diff
@@ -38,8 +38,8 @@ export const Portfolio: React.FC<PortfolioProps> = ({
   onTableChange,
 }) => {
   return (
-    <Box sx={{ pb: { xs: 21.25, sm: 28.28 } }}>
-      <Box sx={{ pt: 12.75 }}>
+    <Box>
+      <Box>
```
Comment (haveanicedavid) [inline web-registry/src/components/organisms/Portfolio.tsx:42]: 👍

**Review by haveanicedavid — APPROVED** [review]
lgtm

**Review by victordelval — APPROVED** [review]
Nice work!

**Comment by flagrede** [comment]
@erikalogie No design changes in this PR but if you want to try out the separate URLs for public and private profile: https://deploy-preview-1549--regen-registry.netlify.app/

**Comment by erikalogie** [comment]
@flagrede The separate urls being "/ecocredits/portfolio" vs "ecocredits/accounts/accountaddress/portfolio"?

This is working well, I wonder though if we want to change this url structure in the future, maybe we should have a separate task for that. For example on OpenSea the public profile is at /accountaddress and the tabs at /accountaddress?tab=created. The private portfolio is at /account and the tabs are at /account?tab=created. This is way simpler than our current structure..

We could have /account as the user's private profile page (perhaps redirects to /account/portfolio) and /accountaddress for the user's public profile page that redirects to /accountaddress/portfolio. Maybe @blushi wants to we in here as well.

**Comment by flagrede** [comment]
@erikalogie Yes portfolio and all other tabs also.
If we want to change URLs later on, it should be reasonably straightforward. I will merge this as it is needed for another story but happy to handle further URL changes in another PR.

**Comment by erikalogie** [comment]
> @erikalogie Yes portfolio and all other tabs also. If we want to change URLs later on, it should be reasonably straightforward. I will merge this as it is needed for another story but happy to handle further URL changes in another PR.

Ok great, I opened a separate task for this.

**Comment by flagrede** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**